static/js/work/controllers/work.controller.js

```diff
@@ -0,0 +1,60 @@
+//__author__ = 'lucamatsumoto'
+
+(function () {
+  'use strict';
+
+	angular
+	    .module('crowdsource.work.controllers', ['ngAnimate', 'ngSanitize', 'mgcrea.ngStrap', 'ngFileUpload'])
+        .filter('bytes', function() {
+            return function(bytes, precision) {
+                if (isNaN(parseFloat(bytes)) || !isFinite(bytes)) return '-';
+                if (typeof precision === 'undefined') precision = 1;
+                var units = ['bytes', 'kB', 'MB', 'GB', 'TB', 'PB'],
+                    number = Math.floor(Math.log(bytes) / Math.log(1024));
+                return (bytes / Math.pow(1024, Math.floor(number))).toFixed(precision) +  ' ' + units[number];
+            }
+        })
+	    .controller('macroWorkSubmitController', macroWorkSubmitController);
+
+	macroWorkSubmitController.$inject = ['$scope', '$modal', '$mdToast', '$http', '$timeout', 'Upload', 'WorkService'];
+
+    function macroWorkSubmitController($scope, $modal, $mdToast, $http, $timeout, Upload, WorkService) { 
+        $scope.displayedCollection = [];
+        $scope.rowCollection=[];
+            
+        $scope.removeRow = function removeRow(row) {
+            var index = $scope.rowCollection.indexOf(row);
+            if (index !== -1) {
+                WorkService.removeFile(row.id)
+                $scope.rowCollection.splice(index, 1);
+            }
+        }
+
+        $scope.taskInfo = "Task information will be displayed here...";
+
+        $scope.$watch('files_to_upload', function () {
+            $scope.upload($scope.files_to_upload);
```

Comment:
a watch should be accompanied by checking newVal !== oldVal, this will trigger on every load otherwise